### PR TITLE
Add option to cascade table truncation in `operations#recreate_all()`.

### PR DESCRIPTION
**Pull request by stuarthadfieldglobality**
Occasionally we will have circumstances where there are cyclic foreign
key constraints on tables.

For instance, a 1 to many relationship between UserIdentity and User.
UserIdentity has a `created_by`, which is a f.k. on `user.id`
User has a `user_identity_id`, which is a f.k. on `user_identity.id`

SqlAlchemy recommends naming Foreign Keys to get around cyclic dependencies,
but the resolution of dependent tables is still not perfect. It so happens
that sometimes the order MetaData.sorted_tables returns tables in is partially
alphabetical. In this instance, calling `.recreate_all()` in test suites
will yield a ForeignKeyViolation, because there are still references to the
table you are trying to truncate in other tables.

In our example above, the reversed sorted tables (i.e. order of deletion)
indicated UserIdentity was first, and then User. You cannot delete a
UserIdentity whilst a User still references it in the `user_identity_id` field,
thus the operation fails.

In this instance, we want to force a cascading delete. We can achieve this
by passing in the flag in this commit to `recreate_all()`. This is a faster
alternative to dropping and recreating all the tables from scratch in between
test runs.

### CODE REVIEW

File: microcosm_postgres/operations.py [hunk 1]
```diff
@@ -93,7 +92,12 @@ def recreate_all(graph):
     connection = graph.postgres.connect()
     transaction = connection.begin()
     for table in reversed(_metadata.sorted_tables):
-        connection.execute(table.delete())
+        # Occasionally SqlAlchemy can't resolve the order in which to truncate records, due to cyclic
+        # Foreign Key constraints. We can use this flag to ensure truncation cascades between records.
+        if cascade_truncation:
+            connection.execute(f"TRUNCATE TABLE \"{table}\" CASCADE")
```
Comment (GabeLambda) [inline microcosm_postgres/operations.py:98]: I think you can use `public.{table}` for a similar effect, if you want to avoid namespace clashes.

**Review by GabeLambda — APPROVED** [review]

**Comment by GabeLambda** [comment]
You can also add a test here to simulate that cyclical dependency, and make sure this works.

**Comment by stuarthadfieldglobality** [comment]
I'm not sure how I can simulate the cyclical dependency.

The tables for tests are created at runtime of the tests, there's no after-the-fact addition of ForeignKeys.

Right now we have the notion of "Employee" and "Company". Supposing I wanted to accurately simulate the case we introduced this change for, I could have a "EmployeeIdentity" table.

The EmployeeIdentity could have a FK on employee.id, and Employee could have a FK on employee_identity.id, but because the one table will never exist before the other, we can't create that cyclic dependency.

Trying to accomplish the above simply yields:
```
sqlalchemy.exc.NoReferencedTableError: Foreign key associated with column 'employee.employee_identity_id' could not find table 'employee_idenity' with which to generate a foreign key to target column 'id'
```

I'm not sure of any other "synthetic" way I could generate the behaviour required for this case, @GabeLambda ... Any help?

**Comment by stuarthadfieldglobality** [comment]
Closing in favour of https://github.com/globality-corp/ariza/pull/489

**Pull request closed without merging**